clients/clients-integration-tests/src/test/java/org/apache/kafka/clients/CreateTopicsRequestWithPolicyTest.java

```diff
@@ -0,0 +1,236 @@
+                        throw new PolicyViolationException("RetentionMs should be less than 5000ms if partitions > 10");
+                    }
+                } else {
+                    assertTrue(requestMetadata.configs().isEmpty(), "Topic configs should be empty, but it is " + requestMetadata.configs());
+                }
+            } else {
+                assertNull(requestMetadata.numPartitions(), "numPartitions should be null, but it is " + requestMetadata.numPartitions());
+                assertNull(requestMetadata.replicationFactor(), "replicationFactor should be null, but it is " + requestMetadata.replicationFactor());
+                assertNotNull(requestMetadata.replicasAssignments(), "replicasAssignments should not be null, but it is " + requestMetadata.replicasAssignments());
+
+                requestMetadata.replicasAssignments().forEach((partitionId, replicas) -> {
+                    if (replicas.size() < 2) {
+                        throw new PolicyViolationException("Topic partitions should have at least 2 replicas, received " +
+                                replicas.size() + " for partition " + partitionId);
+                    }
+                });
+            }
+        }
+
+        @Override
+        public void close() {
+            closed = true;
+        }
+    }
+
+    private void validateValidCreateTopicsRequests(NewTopic topic, Admin admin, boolean validateOnly) throws Exception {
+        admin.createTopics(
+            List.of(topic),
+            new CreateTopicsOptions().validateOnly(validateOnly)
+        ).all().get();
+    }
+
+    @ClusterTest
+    public void testValidCreateTopicsRequests(ClusterInstance cluster) throws Exception {
+        try (Admin admin = cluster.admin()) {
+            cluster.createTopic("topic1", 5, (short) 1);
+
+            validateValidCreateTopicsRequests(
+                new NewTopic("topic2", 5, (short) 3),
+                admin,
+                true
+            );
+
+            validateValidCreateTopicsRequests(
+                new NewTopic("topic3", 11, (short) 2)
+                    .configs(Map.of(TopicConfig.RETENTION_MS_CONFIG, "4999")),
+                admin,
+                true
+            );
+
+            validateValidCreateTopicsRequests(
+                new NewTopic("topic4", Map.of(
+                    0, Arrays.asList(1, 0),
+                    1, Arrays.asList(0, 1)
+                )),
+                admin,
+                false
+            );
+        }
+    }
+
+    private void validateErrorCreateTopicsRequests(NewTopic topic, Admin admin, boolean validateOnly, Class<? extends Throwable> expectedExceptionClass, String expectedErrorMessage) {
+        ExecutionException exception = assertThrows(ExecutionException.class, () ->
+                admin.createTopics(List.of(topic), new CreateTopicsOptions().validateOnly(validateOnly)).all().get());
+        assertInstanceOf(expectedExceptionClass, exception.getCause());
```

Comment:
Could you use a stricter check here, use `assertEquals` instead of `assertInstanceOf`?
```suggestion
       assertEquals(
                expectedExceptionClass,
                exception.getCause(), 
                "Expected " + expectedExceptionClass.getSimpleName() + ", but got " + exception.getCause().getSimpleName()
            );
```

Reply:
Thanks @m1a2st  for the review! I have added the enhanced assertion message to make the output clearer as suggested. However, using `assertEquals` would require creating a addtional instance which may be unnecessary? since we have already verified the exception type and error message here.

Reply:
`assertInstanceOf` allows the subclasses, so it's not a strict check. By contrast, `assertEquals(expectedExceptionClass, exception.getCause().getClass());` is a strict check.

Reply:
Updated. Thanks @m1a2st @chia7712